### PR TITLE
Restructure the kernel-pod.yaml.j2 template

### PR DESCRIPTION
**Pull request by JorisJamers**
As talked about in issue #749 there was a need to document the conditional and unconditional volumeMounts and volumes. 

This should be documented?

### CODE REVIEW

**Comment by kevin-bates** [comment]
@JorisJamers - thank you for the contribution.  Yes, we should probably document this, probably as a subsection at the end of this (parent) section: https://jupyter-enterprise-gateway.readthedocs.io/en/latest/kernel-kubernetes.html#kubernetes-kernel-instances

Seems like a `Mounting volumes` section would be very helpful where we show both "unconditional" mount adjustments, along with those that might vary per user.  You could probably use the examples from @esevan's PR in that portion.  

If you're not comfortable including doc changes, we can create an issue with the appropriate information instead.

**Comment by JorisJamers** [comment]
Started adding some documentation in the appropriate section, feel free to add / change / move the documentation at your will @kevin-bates, used the `/dev/shm` example because I'm most familiar with that use case.

File: docs/source/kernel-kubernetes.md [hunk 1]
```diff
@@ -382,18 +382,62 @@ spec:
...
 1. Kernel pods have restart policies of `Never`.  This is because the Jupyter framework already has built-in logic for auto-restarting failed kernels and any other restart policy would likely interfere with the built-in behaviors.
 1. The parameters to the launcher that is built into the image are communicated via environment variables as noted in the `env:` section above.
 
+### Unconditional Volume Mounts
+Unconditional volume mounts can be added in the `kernel-pod.yaml.j2` template. An example for these unconditional can be the docker shared memory. For some I/O jobs the pod will need more than the default `64mb` of shared memory on the `/dev/shm` path.
```
Comment (kevin-bates) [inline docs/source/kernel-kubernetes.md]: I think we're going to want some examples for how to use the the conditional logic.  This is great for the unconditional discussion.  We can add the conditional discussion in a follow-on PR if you prefer.

How about updating the second sentence to some like this... _An example of these unconditional volume mounts can be found when extending docker shared memory._

**Comment by JorisJamers** [comment]
Sentence change happened! Someone might add in to the conditional volume mounts because I'm not too familiar with that concept.

**Pull request merged**